rebirthdb/gevent_net/net_gevent.py

```diff
@@ -15,29 +15,31 @@
 # This file incorporates work covered by the following copyright:
 # Copyright 2010-2016 RethinkDB, all rights reserved.
 
-import errno
-import ssl
 import struct
 
 import gevent
-import gevent.socket as socket
```

Comment:
This is the reason I noted waiting on tests below. The documentation for this class says it will only block one greenlet and that is essential to being async. I would rather wait for tests. Otherwise socket wrapper should work like connection where it operates on an internal instance object which can be overridden in the subclass constructor. Long term this pattern is a good place for metaclass factories.